### PR TITLE
feat: LW-10710 generate shared keys flow

### PR DESCRIPTION
**Pull request by szymonmaslowski**
# Checklist

- [x] JIRA - [\<link>](https://input-output.atlassian.net/browse/LW-10710)
- [ ] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

UI flow for keys generation ([figma](https://www.figma.com/design/RPN3V4Et7hMbXkV5ZIF75i/[deprecated]-Lace-Browser---Full-view-UI-Design?node-id=42840-488135&t=JjKiRmmziONSnUAI-0)). Fake key is being "generated" for now.

## Testing

- For now, as the shared key is fake that key is being forgotten every time the modal is closed. This is intentional until we have a real key generation in place.
- For the same reason password is not verified yet. It will be addressed in the real key generation implementation (same ticket, next PR).

## Screenshots

image
image
image
image
image
image
<br />
<br />

### Password error state - reproducible only in storybook (core package, yarn dev, story: Shared Wallets > Flows > Add shared wallet - keys derivation > Enter password - error message)

image

### CODE REVIEW

**Comment by pczeglik-iohk** [comment]
# Allure Report
[`allure-report-publisher`](https://github.com/andrcuns/allure-report-publisher) generated test report!



**smokeTests**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/smoke/linux/chrome/6051/9799458036/index.html) for [043f8b70](https://github.com/input-output-hk/lace/pull/1272/commits/043f8b705f86076592f35792d48074121e406de9)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 33     | 0      | 0       | 0     | 33    | ✅     |

File: apps/browser-extension-wallet/src/views/browser-view/features/shared-wallet/SharedWallet.tsx [hunk 1]
```diff
@@ -1,28 +1,47 @@
...
     })();
-  }, [walletRepository]);
+  }, [walletManager.activeWalletId$, walletRepository.wallets$]);
 
-  const sharedKeys = 'addr_shared_vksdhgfsft578s6tf68tdsf,stake_shared_vkgyufieus65cuv76s5vrs7';
+  const generateKeys = async () => {
+    const keys = 'addr_shared_vksdhgfsft578s6tf68tdsf,stake_shared_vkgyufieus65cuv76s5vrs7';
+    setSharedKeys(keys);
+    return keys;
+  };
```
Comment (szymonmaslowski) [inline apps/browser-extension-wallet/src/views/browser-view/features/shared-wallet/SharedWallet.tsx:44]: A fake `generateKeys` function for now. I will create the real one in the next PR.

File: apps/browser-extension-wallet/src/views/browser-view/features/shared-wallet/SharedWallet.tsx [hunk 2]
```diff
@@ -34,16 +53,31 @@ export const SharedWallet = (): JSX.Element => {
...
               activeWalletName={walletInfo?.name || ''}
-              initialWalletName={initialWalletName}
-              navigateToAppHome={() => setBackgroundPage()}
+              activeWalletType={activeWalletType}
+              generateKeys={generateKeys}
               navigateToParentFlow={() => history.push(walletRoutePaths.sharedWallet.root)}
             />
           )}
         />
+        {sharedKeys && (
```
Comment (szymonmaslowski) [inline apps/browser-extension-wallet/src/views/browser-view/features/shared-wallet/SharedWallet.tsx:66]: If there is no keys generated yet, then create and import routes should be inactive

File: packages/core/src/shared-wallets/add-shared-wallet/generate-keys-flow/Store/machine.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  GenerateSharedKeysStep,
+  StateCopyKeys,
+  StateEnterPassword,
+  stateCopyKeys,
+  stateEnterPassword,
+} from './state';
+
+export enum ActionType {
+  Back = 'Back',
+  CloseFlow = 'CLoseFlow',
```
Comment (vetalcore) [inline packages/core/src/shared-wallets/add-shared-wallet/generate-keys-flow/Store/machine.ts:13]:
```suggestion
  CloseFlow = 'CloseFlow',
```

**Review by vetalcore — APPROVED** [review]

**Review by shawnbusuttil — APPROVED** [review]

File: packages/core/src/shared-wallets/add-shared-wallet/generate-keys-flow/Store/machine.ts (hunk 3 above)
Comment (szymonmaslowski) [reply packages/core/src/shared-wallets/add-shared-wallet/generate-keys-flow/Store/machine.ts:13]: Let me fix that in the next PR.

**Pull request merged**